open_instruct/test_utils.py

```diff
@@ -0,0 +1,100 @@
+
+    def assert_args_type(self):
+        dataset_mixer = {
+            "HuggingFaceH4/testing_alpaca_small": 0.5,
+            "HuggingFaceH4/testing_self_instruct_small": 0.3,
+            "HuggingFaceH4/testing_codealpaca_small": 0.2,
+        }
+        _ = FlatArguments(dataset_mixer, columns_to_keep=["prompt", "completion"])
+
+    def test_loading_data_args(self):
+        dataset_mixer = {
+            "HuggingFaceH4/testing_alpaca_small": 0.5,
+            "HuggingFaceH4/testing_self_instruct_small": 0.3,
+            "HuggingFaceH4/testing_codealpaca_small": 0.2,
+        }
+        datasets = get_datasets(dataset_mixer, columns_to_keep=["prompt", "completion"])
+        self.assertEqual(len(datasets["train"]), 100)
+        self.assertEqual(len(datasets["test"]), 300)
+
+    def test_loading_with_unit_fractions(self):
+        dataset_mixer = {
+            "HuggingFaceH4/testing_alpaca_small": 1.0,
+            "HuggingFaceH4/testing_self_instruct_small": 1.0,
+            "HuggingFaceH4/testing_codealpaca_small": 1.0,
+        }
+        datasets = get_datasets(dataset_mixer, columns_to_keep=["prompt", "completion"])
+        self.assertEqual(len(datasets["train"]), 300)
+        self.assertEqual(len(datasets["test"]), 300)
+
+    def test_loading_with_fractions_greater_than_unity(self):
+        dataset_mixer = {
+            "HuggingFaceH4/testing_alpaca_small": 0.7,
+            "HuggingFaceH4/testing_self_instruct_small": 0.4,
+        }
+        datasets = get_datasets(dataset_mixer, columns_to_keep=["prompt", "completion"])
+        self.assertEqual(len(datasets["train"]), 70 + 40)
+        self.assertEqual(len(datasets["test"]), 200)
+
+    def test_loading_fails_with_negative_fractions(self):
+        dataset_mixer = {
+            "HuggingFaceH4/testing_alpaca_small": 0.7,
+            "HuggingFaceH4/testing_self_instruct_small": -0.3,
+        }
+        with pytest.raises(ValueError, match=r"Dataset fractions / lengths cannot be negative."):
+            get_datasets(dataset_mixer, columns_to_keep=["prompt", "completion"])
+
+    def test_loading_single_split_with_unit_fractions(self):
+        dataset_mixer = {
+            "HuggingFaceH4/testing_alpaca_small": 1.0,
+        }
+        datasets = get_datasets(dataset_mixer, splits=["test"], columns_to_keep=["prompt", "completion"])
+        self.assertEqual(len(datasets["test"]), 100)
+        self.assertRaises(KeyError, lambda: datasets["train"])
+
+
+# useful for checking if public datasets are still available
+# class CheckTuluDatasetsTest(unittest.TestCase):
+#     """
+#     Try to rebuild Tulu from public sources
+#     """
+
+#     def test_loading_tulu(self):
+#         dataset_mixer = {
+#             "natolambert/tulu-v2-sft-mixture-flan": 50000,
+#             "natolambert/tulu-v2-sft-mixture-cot": 49747,
+#             "allenai/openassistant-guanaco-reformatted": 7708,  # not exact subset
+#             "Vtuber-plan/sharegpt-cleaned": 114046,
```

Comment:
Cool, I didn't know this exists. Do they have some documentation on what are cleaned?